Make promote-menhir failing, cannot build parser

Hello, I'm trying to re-generate the parser. Eventually I want to modify the parser, but I'm running into errors trying to build it without even modifying it. The full set of commands I've run is as follows (from the root of the repo)
```
./configure
make
make install
make promote-menhir
```
The first three succeed without issue. The final step is failing with the following error
```
menhir --lalr --explain --dump --require-aliases --strict -lg 1 -la 1 --unused-token COMMENT --unused-token DOCSTRING --unused-token EOL --unused-token GREATERRBRACKET  --infer --ocamlc "./boot/ocamlrun ./boot/ocamlc -nostdlib -I ./boot -use-prims runtime/primitives " --fixed-exception --table --strategy simplified  parsing/parser.mly
File "parsing/parser.mly", line 748, characters 17-27:
Error: Unbound module Location
make: *** [promote-menhir] Error 1
```
It seems that menhir is unaware of any of the modules in `parsing/` directory, since deleting all references to `Location` in `parser.mly` results in errors about `Asttypes` being unbound. Am I doing something obviously wrong?

<!--
Welcome to OCaml's Issue tracker!

OCaml's developers use this tracker for bugs and feature requests only, rather
than user support.

If you have questions about *using* OCaml, please ask at
https://discuss.ocaml.org (more people read Discuss than this tracker, and
you'll get confirmation of whether you've really found a bug or need a new
feature).

If your error came from the OCaml package manager, opam, (messages beginning
`[ERROR] The compilation of ...`), please start at
https://github.com/ocaml/opam-repository/issues/new.

Some libraries and tools which used to be part of OCaml are now maintained
separately. Please post questions about Graphics, Num, camlp4, LablTk, CamlDBM
or OCamlTopWin on Discuss or on their respective issue trackers (see [README.adoc](https://github.com/ocaml/ocaml/blob/trunk/README.adoc#separately-maintained-components)
for a full list).
-->

--- a/Makefile.menhir
+++ b/Makefile.menhir
@@ -76,7 +76,7 @@ MENHIRBASICFLAGS := \
 MENHIRFLAGS := \
   $(MENHIRBASICFLAGS) \
   --infer \
-  --ocamlc "$(CAMLC) $(COMPFLAGS)" \
+  --ocamlc "$(CAMLC) $(OC_COMMON_CFLAGS) $(INCLUDES)" \
   --fixed-exception \
   --table \
   --strategy simplified \

